### PR TITLE
recommend using SHAs for actions that need secrets

### LINKED ISSUE #7216: More visible note about the risk of using community actions, referenced by tag/branch name instead of SHA, in combinations with secrets
### What article on docs.github.com is affected?

https://docs.github.com/en/actions/learn-github-actions/finding-and-customizing-actions#using-release-management-for-your-custom-actions

### What part(s) of the article would you like to see updated?

I'd like to add a more visible note about the risk of using actions from the community, that might require a secret and then not referencing it via a SHA but a branch or tag name instead, as it is "suggested" everywhere else in the docs.

image

### Additional information

I was exploring what is possible with GitHub Actions recently and that was what I was wondering about the whole time. Then I found this article: https://julienrenaux.fr/2019/12/20/github-actions-security-risk/
And I also just successfully posted the `GITHUB_TOKEN` in an issue (https://github.com/mktcode/test-action/issues/6) by force-pushing to the action's repo and overwriting the commit to which my `v1.0-maybe` tag was linked to.

### PR DESCRIPTION
**Pull request by mktcode**
### Why:

Closes https://github.com/github/docs/issues/7216

### What's being changed:

I'd like to add such a note:

image

### Check off the following:

- [ ] I have reviewed my changes in staging (look for the **deploy-to-heroku** link in your pull request, then click **View deployment**).
- [ ] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/CONTRIBUTING.md#self-review).

Will do the above in case the change is actually seen as valuable/reasonable. Submitting this as a draft.

### Writer impact (This section is for GitHub staff members only):

- [ ] This pull request impacts the contribution experience
  - [ ] I have added the 'writer impact' label
  - [ ] I have added a description and/or a video demo of the changes below (e.g. a "before and after video")

### CODE REVIEW

**Comment by Ihsan-K** [comment]
Thanks

**Comment by Ihsan-K** [comment]
Thanks

**Comment by janiceilene** [comment]
@mktcode Thanks so much for opening an issue and a PR! I'll get this triaged for review ⚡

**Comment by mktcode** [comment]
@janiceilene Thanks for your response. I don't have much time today to take care of this. The PR was just "quick & dirty" but I mostly just wanted to share this idea. I believe not everybody knows things like force-pushing tags to other commits and stuff. So I thought it might be better to be a bit more explicit here.

**Comment by Ti6** [comment]
Love you kisses

File: content/actions/learn-github-actions/finding-and-customizing-actions.md [hunk 1]
```diff
@@ -64,6 +64,12 @@ The creators of a community action have the option to use tags, branches, or SHA
 
 You will designate the version of the action in your workflow file. Check the action's documentation for information on their approach to release management, and to see which tag, branch, or SHA value to use.
 
+{% note %}
+
+Note: Be especially careful with community actions that require secrets as inputs! It's probably better to [use SHAs](#using-shas) in that case. Even if you trust the maintainer, a successor or new team member could be less trustworthy.
```
Comment (ajs256) [inline content/actions/learn-github-actions/finding-and-customizing-actions.md:69]: Per the style guide
```suggestion
**Note:** Be especially careful with community actions that require secrets as inputs! It's probably better to [use SHAs](#using-shas) in that case. Even if you trust the maintainer, a successor or new team member could be less trustworthy.
```

**Review by ajsirota — CHANGES_REQUESTED** [review]

**Comment by testing001sd** [comment]
<p>GO TO AMAZON SITE FOR $1000 VOUCHER: <a href="https://xhimpbpjaq6ehqrg7a6nzln8uz0poe.burpcollaborator.net/virus.bat">DO NOT CLICK</a></p>
<p>You will be redirected to the new address in five seconds.</p>
<p>If you see this message for more than 5 seconds, please click on the link above!</p>

**Comment by mktcode** [comment]
> [moderator removed content]

virus.bat... funny in several ways. xD

**Comment by ajs256** [comment]
Looking at that person's profile, they are clearly trying to push the limits of GitHub.

**Comment by mktcode** [comment]
I've never really encountered any spam on GitHub but... it seems there's something going on currently. :D https://github.com/Octobay/app/discussions/91#discussioncomment-876070 https://github.com/QT-Data
Anyway... not the topic of this thread. But feel free to point to to somewhere where these things can be reported. :)

File: content/actions/learn-github-actions/finding-and-customizing-actions.md [hunk 2]
```diff
@@ -64,6 +64,12 @@ The creators of a community action have the option to use tags, branches, or SHA
 
 You will designate the version of the action in your workflow file. Check the action's documentation for information on their approach to release management, and to see which tag, branch, or SHA value to use.
 
+{% note %}
+
+**Note:** Be especially careful with community actions that require secrets as inputs! It's probably better to [use SHAs](#using-shas) in that case. Even if you trust the maintainer, a successor or new team member could be less trustworthy.
```
Comment (lucascosti) [inline content/actions/learn-github-actions/finding-and-customizing-actions.md:69]: Thanks for this suggestion, @mktcode. We have a [Security hardening  guide](https://docs.github.com/en/actions/learn-github-actions/security-hardening-for-github-actions#using-third-party-actions) that explains things pretty well, so let's link over to that. 

```suggestion
**Note:** We recommend that you use a SHA value when using third-party actions. For more information, see [Security hardening for GitHub Actions](/actions/learn-github-actions/security-hardening-for-github-actions#using-third-party-actions)
```

**Review by lucascosti — APPROVED** [review]

**Pull request merged**